tools/pcd_viewer.cpp

```diff
@@ -152,6 +154,18 @@ pcl::search::KdTree<pcl::PointXYZ> search;
 pcl::PCLPointCloud2::Ptr cloud;
 pcl::PointCloud<pcl::PointXYZ>::Ptr xyzcloud;
 
+void
+area_callback(const pcl::visualization::AreaPickingEvent& event, void* /*cookie*/)
+{
+  const auto names = event.getCloudNames();
+
+  for (const std::string& name : names) {
+    const pcl::Indices indices = event.getPointsIndices(name);
+
+    PCL_INFO("Picked %d points from %s \n", indices.size(), name.c_str());
```

Comment:
How about doing something useful with the indices, for example writing the points to PCD files?

Reply:
It seems to require a bit of work. Ie. I need to save the pointclouds in a map as well as handling multiples types of pointclouds (pointcloud2).
At also seems that the indices thats extracted, is not necessarily one-to-one mapping, if there is NaN points in the pointcloud.
Its a bit too much work for me - but we could add it as an issue?